### PR TITLE
[Reviewer: Rob] Sto1117

### PR DESCRIPTION
**Pull request by eleanor-merry**
Rob, can you please review these change to plumb exception handling into Chronos

### CODE REVIEW

File: src/main/globals.cpp [hunk 1]
```diff
@@ -91,6 +92,10 @@ void Globals::update_config()
   set_threads(threads);
   LOG_STATUS("HTTP Threads: %d", threads);
 
+  int ttl = conf_map["exception.max_ttl"].as<int>();
+  set_max_ttl(ttl);
+  LOG_STATUS("Maximum TTL: %d", ttl);
```
Comment (rkd-msw) [inline src/main/globals.cpp]: Can this log say "Maximum post-exception TTL", or something more specific along those lines?

File: src/main/main.cpp [hunk 2]
```diff
@@ -26,21 +29,25 @@ void terminate_handler(int sig)
...
   LOG_BACKTRACE("Signal %d caught", sig);
 
   // Ensure the log files are complete - the core file created by abort() below
   // will trigger the log files to be copied to the diags bundle
   LOG_COMMIT();
 
+  // Check if there's a stored jmp_buf on the thread and handle if there is
+  exception_handler->handle_exception();
+
+  CL_CHRONOS_CRASHED.log(strsignal(sig));
```
Comment (rkd-msw) [inline src/main/main.cpp]: This is now out-of-date, I think - from looking at https://github.com/Metaswitch/chronos/blob/dev/src/include/chronos_pd_definitions.h#L60, we'll drop a 'Chronos is exiting' log several minutes before we exit.

**Comment by rkd-msw** [comment]
Looks good. Two comments, both about logs.

File: src/main/main.cpp (hunk 2 above)
Comment (eleanor-merry) [inline src/main/main.cpp]: Nope, we'll only hit this if it's the second abort, or if there was an exception hit that didn't have setjmp called on the thread. We won't print this on the first exception

Comment (rkd-msw) [inline src/main/main.cpp]: Sounds good, but does that mean our logging is now less useful? For example, should we be logging a "we ht an exception and will exit in %d seconds" ENT log? (If that's complicated, I'm OK with just raising an issue to cover it.)

**Pull request merged**